istio/resources/values-template.yaml

```diff
@@ -1,445 +1,193 @@
-# -- Overrides the chart's name. Used to change the infix in the resource names.
-nameOverride: null
-
-# -- Overrides the chart's computed fullname. Used to change the full prefix of
-# resource names.
-fullnameOverride: null
-
-## Global properties for image pulling override the values defined under `image.registry` and `configReloader.image.registry`.
-## If you want to override only one image registry, use the specific fields but if you want to override them all, use `global.image.registry`
-global:
-  image:
-    # -- Global image registry to use if it needs to be overriden for some specific use cases (e.g local registries, custom images, ...)
-    registry: ""
-
-    # -- Optional set of global image pull secrets.
-    pullSecrets: []
-
-  # -- Security context to apply to the Grafana Agent pod.
-  podSecurityContext: {}
-
-crds:
-  # -- Whether to install CRDs for monitoring.
-  create: true
-
-# Various agent settings.
-agent:
-  # -- Mode to run Grafana Agent in. Can be "flow" or "static".
-  mode: 'flow'
-  configMap:
-    # -- Create a new ConfigMap for the config file.
-    create: true
-    # -- Content to assign to the new ConfigMap.  This is passed into `tpl` allowing for templating from values.
-    content: |
-      discovery.kubernetes "istiod" {
-        role = "endpoints"
-
-        namespaces {
-          names = ["istio-system"]
-        }
+cluster:
+  name: minikube
+
+externalServices:
+  prometheus:
+    host: "$PROM_URL"
+    basicAuth:
+      username: "$PROM_USER"
+      password: "$PROM_PASS"
+  loki:
+    host: "$LOKI_URL"
+    basicAuth:
+      username: "$LOKI_USER"
+      password: "$LOKI_PASS"
+
+logs:
+  enabled: true
+  pod_logs:
+    enabled: true
+
+extraConfig: |-
+  discovery.relabel "istio_proxy_metrics" {
+    targets = discovery.kubernetes.pods.targets
+
+    rule {
+      action        = "keep"
+      source_labels = ["__meta_kubernetes_pod_container_name"]
+      regex         = "istio-proxy.*"
+    }
+    rule {
+      source_labels = ["__meta_kubernetes_pod_annotation_prometheus_io_port", "__meta_kubernetes_pod_ip"]
+      regex         = "(\\d+);(([A-Fa-f0-9]{1,4}::?){1,7}[A-Fa-f0-9]{1,4})"
+      target_label  = "__address__"
+      replacement   = "[$2]:$1"
+    }
+    rule {
+      source_labels = ["__meta_kubernetes_pod_annotation_prometheus_io_port", "__meta_kubernetes_pod_ip"]
+      regex         = "(\\d+);((([0-9]+?)(\\.|$)){4})"
+      target_label  = "__address__"
+      replacement   = "$2:$1"
+    }
+    rule {
+      target_label = "job"
+      replacement = "integrations/istio"
+    }
+    rule {
+      target_label  = "cluster"
+      replacement   = "minikube"
+    }
```

Comment:
Same comment as with Loki.. The helm chart should be requiring a cluster already. In this case you can actually write to [this relabel rule](https://github.com/grafana/k8s-monitoring-helm/blob/main/charts/k8s-monitoring/templates/agent_config/_metrics_service.river.txt#L8-L14) rather than the remote_write component as the last link in your chain.